Performance/QPSBenchmark/Sources/QPSBenchmark/Runtime/BenchmarkServiceImpl.swift

```diff
@@ -0,0 +1,111 @@
+  /// One request followed by one response.
+  /// The server returns the client payload as-is.
+  func unaryCall(request: Grpc_Testing_SimpleRequest,
+                 context: StatusOnlyCallContext) -> EventLoopFuture<Grpc_Testing_SimpleResponse> {
+    do {
+      return context.eventLoop
+        .makeSucceededFuture(try AsyncQPSServerImpl.processSimpleRPC(request: request))
+    } catch {
+      return context.eventLoop.makeFailedFuture(error)
+    }
+  }
+
+  /// Repeated sequence of one request followed by one response.
+  /// Should be called streaming ping-pong
+  /// The server returns the client payload as-is on each response
+  func streamingCall(context: StreamingResponseCallContext<Grpc_Testing_SimpleResponse>)
+    -> EventLoopFuture<(StreamEvent<Grpc_Testing_SimpleRequest>) -> Void> {
+    context.logger.warning("streamingCall not implemented yet")
+    return context.eventLoop.makeFailedFuture(GRPCStatus(
+      code: GRPCStatus.Code.unimplemented,
+      message: "Not implemented"
+    ))
+  }
+
+  /// Single-sided unbounded streaming from client to server
+  /// The server returns the client payload as-is once the client does WritesDone
+  func streamingFromClient(context: UnaryResponseCallContext<Grpc_Testing_SimpleResponse>)
+    -> EventLoopFuture<(StreamEvent<Grpc_Testing_SimpleRequest>) -> Void> {
+    context.logger.warning("streamingFromClient not implemented yet")
+    return context.eventLoop.makeFailedFuture(GRPCStatus(
+      code: GRPCStatus.Code.unimplemented,
+      message: "Not implemented"
+    ))
+  }
+
+  /// Single-sided unbounded streaming from server to client
+  /// The server repeatedly returns the client payload as-is
+  func streamingFromServer(
+    request: Grpc_Testing_SimpleRequest,
+    context: StreamingResponseCallContext<Grpc_Testing_SimpleResponse>
+  ) -> EventLoopFuture<GRPCStatus> {
+    context.logger.warning("streamingFromServer not implemented yet")
+    return context.eventLoop.makeFailedFuture(GRPCStatus(
+      code: GRPCStatus.Code.unimplemented,
+      message: "Not implemented"
+    ))
+  }
+
+  /// Two-sided unbounded streaming between server to client
+  /// Both sides send the content of their own choice to the other
+  func streamingBothWays(context: StreamingResponseCallContext<Grpc_Testing_SimpleResponse>)
+    -> EventLoopFuture<(StreamEvent<Grpc_Testing_SimpleRequest>) -> Void> {
+    context.logger.warning("streamingBothWays not implemented yet")
+    return context.eventLoop.makeFailedFuture(GRPCStatus(
+      code: GRPCStatus.Code.unimplemented,
+      message: "Not implemented"
+    ))
+  }
+
+  /// Make a payload for sending back to the client.
+  private static func makePayload(type: Grpc_Testing_PayloadType,
+                                  size: Int) throws -> Grpc_Testing_Payload {
+    if type != .compressable {
+      // Making a payload which is not compressable is hard - and not implemented in
+      // other implementations too.
```

Comment:
I guess they planned on doing this but never did and didn't want to break API 🤷‍♂️